pkg/helpers.go

```diff
@@ -261,3 +262,65 @@ func CreateFullPath(p string) (*os.File, error) {
 	}
 	return os.Create(p)
 }
+
+// AwaitValidLetsEncryptCertificateNTimes do maxTimes attempts until it gets a valid Let's Encrypt certificate.
+func AwaitValidLetsEncryptCertificateNTimes(domain string, dryRun bool, maxTimes int) (bool, error) {
+	log.Println("AwaitValidTLSCertificateNTimes called")
+	if dryRun {
+		log.Printf("[#99] Dry-run mode, AwaitValidTLSCertificateNTimes skipped.")
+		return true, nil
+	}
+	for i := 0; i < maxTimes; i++ {
+		validCertificate, err := IsLetsEncryptCertificateDomain(domain)
+		if err != nil {
+			return false, err
```

Comment:
the test is not if the domain is right, if if is valid certificate. `Curl -k` vs `plain curl`

Reply:
I think you went a bit too deep on this, the idea is to fail when the certificate is not valid. Broken Lock. 


As we allow insecure and it seems to be global.

Reply:
I didn't get it, could you re-phase..?

Reply:
From curl documentation, it would better than me explaining. 

```
       -k, --insecure
              (TLS) By default, every SSL connection curl makes is verified to be secure. This option allows curl to proceed and operate even
              for server connections otherwise considered insecure.

              The server connection is verified by making sure the server's certificate contains the right name and verifies successfully using
              the cert store.

              See this online resource for further details:
               https://curl.se/docs/sslcerts.html

              WARNING: this makes the transfer insecure.

              Example:
               curl --insecure https://example.com

              See also --proxy-insecure and --cacert. 
```

Reply:
This is a broken lock:
![image](https://user-images.githubusercontent.com/53096417/189443563-28a8284d-db7d-4ad7-b01f-935b2b3adfc6.png)

Reply:
This is a valid lock: 
![image](https://user-images.githubusercontent.com/53096417/189443673-271b996b-3f79-4460-9e5f-64277739a9ee.png)


Can you see the difference on the browser?  there is a warning on the address bar..

Reply:
Our current client is working with `--insecure` enabled, meaning we don't validate the cert against a valid cert-chain of trust. It is nothing to do with `let's encrypt` we can mock and "fake be it" and still have an invalid certificate. 


It is a bultin behavior of any http lib, that supports https to handle the cert check.

Reply:
The code in this PR diverges to this direction, creating a custom cert validation rule with limited scope, and not what was requested. I send an invite to a time in Monday to review together and understand better the intent of this PR.